### ISSUE TITLE
[Bug][a11y]: indeterminate checkbox input use indeterminate property rather than aria-checked="mixed"

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3266]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Checkbox

### Expected behavior

Automated testing tools like aXe throw a conformance error when an `aria-checked` attribute is assigned to an HTMLInputElement element, because we should be using the native HTML `checked` attribute or the `indeterminate` property.

We should set the [HTMLInputElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement) object's `indeterminate` property via JavaScript (it cannot be set using an HTML attribute).

### Actual behavior

Automated testing tools like aXe throw a conformance error when an `aria-checked` attribute is assigned to an  [HTMLInputElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement)  element, because we should be using the native HTML `checked` attribute or the `indeterminate` property.

### What browsers are you seeing the problem in?

Firefox, Chrome, Safari, Microsoft Edge

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/checkbox/
2. Open aXe DevTools Extension in the Developer Tools panel (I can be acquired here: https://docs.deque.com/devtools-html/4.0.0/en/devtools-extension#installation)
3. Run "Scan ALL my page"
4. Note that  aXe DevTools reports 7 "Elements must only use allowed ARIA attributes" errors for the indeterminate checkboxes that include `aria-checked="mixed"` attribute.
